From ece8e55301e7627734fc6147dc308e555e14e073 Mon Sep 17 00:00:00 2001
From: RajaniRanjan <rajani.ranjan@intel.com>
Date: Tue, 2 Nov 2021 09:28:48 +0530
Subject: [PATCH] enabling long press on the touch screen

enabling long press touch features to take video in wechat app

Tracked-On:OAM-99821
Signed-off-by: Rajani Ranjan <rajani.ranjan@intel.com>
---
 client/src/main.c | 66 +++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 64 insertions(+), 2 deletions(-)

diff --git a/client/src/main.c b/client/src/main.c
index c892fe3c..fd80333e 100644
--- a/client/src/main.c
+++ b/client/src/main.c
@@ -965,7 +965,38 @@ int eventFilter(void * userdata, SDL_Event * event)
       }
       break;
 
-    case SDL_MOUSEBUTTONDOWN:
+      case  SDL_FINGERDOWN:
+#ifdef USE_INTELVTOUCH
+      vinput_touch(PRESS,state.fullscreen, event->tfinger.x, event->tfinger.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x);
+#else
+      if (!spice_running)
+      {
+        if(params.useSpiceInput && !spice_ready())
+        {
+          if (!spice_connect(params.spiceHost, params.spicePort, "") && !spice_ready() && !spice_process())
+            DEBUG_ERROR("SDL_MOUSEMOTION: reconnect mouse");
+          else
+            spice_running = true;
+        }
+      }
+      if (state.ignoreInput || !params.useSpiceInput)
+        break;
+
+      // The SPICE protocol doesn't support more than a standard PS/2 3 button mouse
+      if (event->button.button > 3)
+        break;
+      if (
+        !spice_mouse_position(event->button.x, event->button.y) ||
+        !spice_mouse_press(event->button.button)
+      )
+      {
+        DEBUG_ERROR("SDL_MOUSEBUTTONDOWN: failed to send message");
+        break;
+      }
+#endif
+      break;
+
+      case SDL_MOUSEBUTTONDOWN:
 #ifdef USE_INTELVTOUCH
       vinput_touch(PRESS,state.fullscreen, event->button.x, event->button.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x);
 #else
@@ -996,7 +1027,38 @@ int eventFilter(void * userdata, SDL_Event * event)
 #endif
       break;
 
-    case SDL_MOUSEBUTTONUP:
+      case  SDL_FINGERUP:
+#ifdef USE_INTELVTOUCH
+      vinput_touch(RELEASE,state.fullscreen, event->tfinger.x, event->tfinger.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x);
+#else
+      if (!spice_running)
+      {
+        if(params.useSpiceInput && !spice_ready())
+        {
+          if (!spice_connect(params.spiceHost, params.spicePort, "") && !spice_ready() && !spice_process())
+            DEBUG_ERROR("SDL_MOUSEMOTION: reconnect mouse");
+          else
+            spice_running = true;
+        }
+      }
+      if (state.ignoreInput || !params.useSpiceInput)
+        break;
+
+      // The SPICE protocol doesn't support more than a standard PS/2 3 button mouse
+      if (event->button.button > 3)
+        break;
+      if (
+        !spice_mouse_position(event->button.x, event->button.y) ||
+        !spice_mouse_press(event->button.button)
+      )
+      {
+        DEBUG_ERROR("SDL_MOUSEBUTTONDOWN: failed to send message");
+        break;
+      }
+#endif
+      break;
+
+      case SDL_MOUSEBUTTONUP:
 #ifdef USE_INTELVTOUCH
       vinput_touch(RELEASE,state.fullscreen, event->button.x, event->button.y, state.rotate, state.dstRect.h,state.dstRect.w,state.dstRect.x);
 #else
-- 
2.25.1

